sdk/communication/Azure.Communication.NetworkTraversal/src/CommunicationRelayClient.cs

```diff
@@ -0,0 +1,143 @@
+            : this(endpoint, options.BuildHttpPipeline(tokenCredential), options)
+        { }
+
+        private CommunicationRelayClient(string endpoint, HttpPipeline httpPipeline, CommunicationRelayClientOptions options)
+        {
+            _clientDiagnostics = new ClientDiagnostics(options);
+            RestClient = new CommunicationNetworkTraversalRestClient(_clientDiagnostics, httpPipeline, endpoint, options.ApiVersion);
+        }
+
+        #endregion
+
+        /// <summary>Initializes a new instance of <see cref="CommunicationRelayClient"/> for mocking.</summary>
+        protected CommunicationRelayClient()
+        {
+            _clientDiagnostics = null;
+            RestClient = null;
+        }
+
+        /// <summary>Gets a TURN credential for a <see cref="CommunicationUserIdentifier"/>.</summary>
+        /// <param name="communicationUser">The <see cref="CommunicationUserIdentifier"/> for whom to issue a token.</param>
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        /// <exception cref="RequestFailedException">The server returned an error.</exception>
+        public virtual Response<CommunicationRelayConfiguration> GetRelayConfiguration(CommunicationUserIdentifier communicationUser, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CommunicationRelayClient)}.{nameof(GetRelayConfiguration)}");
+            scope.Start();
+            try
+            {
+                return RestClient.IssueTurnCredentials(communicationUser.Id, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>Asynchronously gets a TURN credential for a <see cref="CommunicationUserIdentifier"/>.</summary>
+        /// <param name="communicationUser">The <see cref="CommunicationUserIdentifier"/> for whom to issue a token.</param>
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        public virtual async Task<Response<CommunicationRelayConfiguration>> GetRelayConfigurationAsync(CommunicationUserIdentifier communicationUser, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CommunicationRelayClient)}.{nameof(GetRelayConfiguration)}");
+            scope.Start();
+            try
+            {
+                return await RestClient.IssueTurnCredentialsAsync(communicationUser.Id, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        private static T AssertNotNull<T>(T argument, string argumentName)
```

Comment:
@pakrym we have these `AssertNotNull` methods in all of our clients. We need them when we need to evaluate the value while passing from one constructor to another so it needs a return like this:
https://github.com/Azure/azure-sdk-for-net/pull/21094/files#diff-dbe38327fce577708241f8eee5f31f74ce25bf15b8815370dd5e1d3297d42f26R27
Does it makes sense to move these to the Arguments class in Azure.Core/Shared?

Reply:
I think that would work.
@heaths, what do you think?

Reply:
It would potentially increase the stack size by 1 and require a noop to drop the return value for every call. Maybe the JIT will optimize it. It may be interesting to benchmark the difference. We have a lot of asserts throughout the code so it could add up if it even makes a practical difference anyway.

Apart from that, I don't see any problem with changing the existing methods.

Even if it does make a difference or there's other reasons, Arguments is already partial so you could have a shared file people could opt into that has a variant like that - maybe something like `IfNotNull` or `CheckNotNull` or something (since asserts, in general, typically do not return anything).